frontend/src/components/common/CommonTable.tsx

```diff
@@ -0,0 +1,374 @@
+
+import pencil from "../../assets/pencil_edit.png";
+
+export type TableTypes = "string" | "number" | "boolean" | "date";
+
+export type ColumnInfoTypes = { Header: string; Type: TableTypes; Key: string };
+
+export type TableProps = {
+  // eslint-disable-next-line
+  data: any[];
+  columnInfo: ColumnInfoTypes[];
+  // eslint-disable-next-line
+  onEdit: (row: any) => any;
+  maxResults: number;
+  isSelectable: boolean;
+};
+
+export const mockData = [
+  {
+    id: 1,
+    title: "taxes",
+    description: "do ur taxes",
+    room: 1,
+    credits: "$4.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 2,
+    title: "more taxes",
+    description: "do ur taxes",
+    room: 1,
+    credits: "$3.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 3,
+    title: "heap of taxes",
+    description: "do ur taxes",
+    room: 2,
+    credits: "$2.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 4,
+    title: "im too lazy",
+    description: "do ur taxes",
+    room: 2,
+    credits: "$4.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 5,
+    title: "t5",
+    description: "do ur taxes",
+    room: 3,
+    credits: "$3.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 6,
+    title: "t6",
+    description: "do ur taxes",
+    room: 3,
+    credits: "$2.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 7,
+    title: "t7",
+    description: "do ur taxes",
+    room: 4,
+    credits: "$3.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 8,
+    title: "t8",
+    description: "do ur taxes",
+    room: 4,
+    credits: "$2.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 9,
+    title: "t9",
+    description: "do ur taxes",
+    room: 5,
+    credits: "$3.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 10,
+    title: "t10",
+    description: "do ur taxes",
+    room: 5,
+    credits: "$2.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 11,
+    title: "t11",
+    description: "do ur taxes",
+    room: 6,
+    credits: "$3.00",
+    due_date: "2022-12-12",
+  },
+  {
+    id: 12,
+    title: "t12",
+    description: "do ur taxes",
+    room: 6,
+    credits: "$2.00",
+    due_date: "2022-12-12",
+  },
+];
+
+export const mockColumns: ColumnInfoTypes[] = [
+  {
+    Header: "Task name",
+    Type: "string",
+    Key: "title",
+  },
+  {
+    Header: "Room Number",
+    Type: "number",
+    Key: "room",
+  },
+  {
+    Header: "Due Date",
+    Type: "date",
+    Key: "due_date",
+  },
+  {
+    Header: "Requested Marillac Bucks",
+    Type: "number",
+    Key: "credits",
+  },
+];
+
+const CommonTable = ({
+  columnInfo,
+  data,
+  isSelectable,
+  maxResults,
+  onEdit,
+}: TableProps): React.ReactElement => {
+  const checkRows: boolean[] = [];
+  Object.keys(data).forEach((item) => {
+    if (Object.prototype.hasOwnProperty.call(data, item)) {
+      checkRows.push(false);
+    }
+  });
+  const [checked, setChecked] = useState(checkRows);
```

Comment:
You should be able to simplify this with the following:

`const [checked, setChecked] = useState(data.map(() => false));`